site/_posts/2020-07-22-securely-implement-oauth-vuejs.md

```diff
@@ -0,0 +1,1213 @@
+        {
+          registration: {
+            data: req.body,
+          },
+        },
+        {
+          headers: {
+            Authorization: process.env.API_KEY,
+          },
+        }
+      ).catch(err=>{
+          console.log(err)
+      })
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+});
+module.exports = router;
+```
+
+To ensure that we are updating the user currently logged in, we extract the token from our server which is done by making a **POST** request to the `oauth/introspect` endpoint; this is similar to what we did in the `user` route. 
+
+Once this request is successful, we make a **PATCH** request to `/api/user/registration` API. If you go through the [User Registration docs](https://fusionauth.io/docs/v1/tech/apis/registrations#update-a-user-registration), you will find that this API accepts both **PUT** and **PATCH** requests but here we are using **PATCH** since we only want to update a single part of the resource and **PATCH** will merge the request parameters into the existing object. 
+
+The `data` to send is stored inside the `registration` object which takes it value from `req.body`. This `registration` represents a user's association with an application. The `data` attribute allows us to store arbitrary data related to a user's registration in an application. We will be sending user data from our Vue app to the Express server via JSON in the body of a **POST** HTTP message.
+
+## Setting user data from Vue
+
+Now that we have created our route for updating user data, let’s create a `text-area` in our Vue app. We will type the user data in that, which will be sent to the server when the `Submit` button is clicked.
+
+In `client/src` create a new file named `Update.vue` and add the following to it:
+
+```html
+<template>
+  <form>
+    <textarea
+      v-model="userData"
+      placeholder="Update FusionAuth user data."
+    ></textarea>
+    <button type="submit" class="button">Submit</button>
+  </form>
+</template>
+<script>
+export default {
+  name: "Update",
+  data() {
+    return {
+      userData: "",
+    };
+  },
+</script>
+<style>
+textarea {
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+}
+button {
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 5px;
+}
+</style>
+```
+
+One of the cool features of Vue is that by using `v-model="userData"` and initializing `userData=""` in the `data()` function, it creates a two-way data binding between `textarea` and `userData`.
+
+We can now access whatever we type in `textarea` in `userData`. You can [read more about it here](https://vuejs.org/v2/guide/forms.html).
+
+Then add this component to `App.vue` . It does not make sense to show this component when the user is not logged in. To hide it, just add `v-if="email"` with this component. Like some of our other components it will check to see if `email` is present or not, i.e., if user is logged in or not.
```

Comment:
The last sentence needs work. It needs a comma after the "like" clause and the "i.e." part should probably be in parenthesis and might not need any of the commas.